airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/SyncCheckConnectionFailure.java

```diff
@@ -70,7 +69,26 @@ public StandardSyncOutput buildFailureOutput() {
                     .withRecordsEmitted(0L)
                     .withBytesEmitted(0L)
                     .withStateMessagesEmitted(0L)
-                    .withRecordsCommitted(0L)));
+                    .withRecordsCommitted(0L)));;
+
+    if (failureOutput.getFailureReason() != null) {
+      syncOutput.setFailures(List.of(failureOutput.getFailureReason().withFailureOrigin(origin)));
+    } else {
+      final StandardCheckConnectionOutput checkOutput = failureOutput.getCheckConnection();
+      final Exception ex = new IllegalArgumentException(checkOutput.getMessage());
+      final FailureReason checkFailureReason = FailureHelper.checkFailure(ex, jobId, attemptId, origin);
+      syncOutput.setFailures(List.of(checkFailureReason));
+    }
```

Comment:
When building the failure for the checks that are executed before syncs, if it failed with a FailureReason from the connector just use that. Otherwise, build a FailureReason just as before.